### PR TITLE
Add sharing status to hovercards

### LINKED ISSUE #6542: show if contact is following back in hovercards
mockup:
hovercard
##

### PR DESCRIPTION
**Pull request by Flaburgan**
This is based on https://github.com/diaspora/diaspora/pull/8316 please merge it first.

Fixes https://github.com/diaspora/diaspora/issues/6542

In the same time, I also fixed the distorted "non-sharing" circle which was looking like this before:
Capture d’écran de 2021-10-31 15-33-06

### CODE REVIEW

File: app/assets/javascripts/app/views/hovercard_view.js [hunk 1]
```diff
@@ -102,14 +107,14 @@ app.views.Hovercard = app.views.Base.extend({
       if( !person || person.length === 0 ) {
         throw new Error("received data is not a person object");
       }
-
+      var personModel = new app.models.Person(person);
+      person.is_sharing = personModel.isSharing();
```
Comment (diaspora-code-review) [inline app/assets/javascripts/app/views/hovercard_view.js:111]: Identifier 'is_sharing' is not in camel case.

File: app/assets/stylesheets/people.scss [hunk 2]
```diff
@@ -45,3 +45,11 @@
     .btn-danger { margin-top: 9px; }
   }
 }
+#sharing_message {
+  &.entypo-check {
+    color: darken($brand-success, 20%);
+  }
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/people.scss:51]: Rule declaration should be followed by an empty line

File: app/assets/stylesheets/people.scss [hunk 3]
```diff
@@ -45,3 +45,11 @@
     .btn-danger { margin-top: 9px; }
   }
 }
+#sharing_message {
```
Comment (diaspora-code-review) [inline app/assets/stylesheets/people.scss:48]: - Avoid using id selectors
- Selector `sharing_message` should be written in lowercase with hyphens

File: app/assets/javascripts/app/views/hovercard_view.js (hunk 1 above)
Comment (Flaburgan) [reply app/assets/javascripts/app/views/hovercard_view.js:111]: The handlebar helper is using this variable name, I don't think it's worth to refactor this.

**Comment by Flaburgan** [comment]
I added some tests, and everything is green \o/
This is ready for review, but please merge https://github.com/diaspora/diaspora/pull/8316 first.

File: app/assets/stylesheets/people.scss [hunk 4]
```diff
@@ -45,3 +47,13 @@
     .btn-danger { margin-top: 9px; }
   }
 }
+
+#sharing_message {
+  &.entypo-check {
+    color: darken($brand-success, 20%);
```
Comment (SuperTux88) [inline app/assets/stylesheets/people.scss:53]: You missed the override for this for the dark theme.

https://github.com/diaspora/diaspora/blob/9fa032e4c91710df072291a26b5344429fd69d65/app/assets/stylesheets/color_themes/_color_theme_override_dark.scss#L58-L60

You moved this so it also selects the sharing state on the hovercard ... but you didn't change the override on the dark theme, which still only selects the profile, so the dark hovercard looks like this now:

image

**Review by SuperTux88 — CHANGES_REQUESTED** [review]

File: app/assets/stylesheets/people.scss (hunk 4 above)
Comment (msph31) [reply app/assets/stylesheets/people.scss:53]: good work supertux88
(btw it saddens me to see "mara" as test subject here):
> image

Comment (Flaburgan) [reply app/assets/stylesheets/people.scss:53]: Nice catch! We should have a checklist on the wiki about what to test while changing the UI: user not logged in, dark theme, mobile... Those are common things which tend to be forgotten.

Comment (Flaburgan) [reply app/assets/stylesheets/people.scss:53]: Fixed. This looks weird to me though, we should only need to set different variables to make the themes work. But this is another topic.

**Review by SuperTux88 — APPROVED** [review]

**Comment by SuperTux88** [comment]
Merged as c67fc4e0f77cd687c314184c33c08b5e3f2594ed

Thx for this feature :cookie:

**Pull request closed without merging**